src/repositories/user/UserRepository.ts

```diff
@@ -0,0 +1,56 @@
+import * as mongoose from 'mongoose';
+import IUserModel from './IUserModel';
+import { userModel } from './UserModel';
+export default class UserRepository {
+
+    public static generateObjectId() {
+        console.log(mongoose.Types.ObjectId());
+        return String(mongoose.Types.ObjectId());
+    }
+
+    private model: mongoose.Model<IUserModel>;
+
+    constructor() {
+        this.model = userModel;
+    }
+
+    public createUser(data: any): Promise<IUserModel> {
+        return this.model.create({
+            ...data,
+            _id: UserRepository.generateObjectId(),
+        });
+    }
+
+    public delete(data: any): any {
+        return this.model.deleteOne(data, (err) => {
+            if (err) {
+                console.log('Error');
+            }
+        });
+    }
+
+    public updateUser(oldData: any, newData: any) {
```

Comment:
make all the repo methods promise based.

Reply:
Done